src/rest_framework_dso/paginator.py

```diff
@@ -0,0 +1,241 @@
+from typing import Callable, Iterable, List, Optional
+
+from django.core.paginator import EmptyPage
+from django.core.paginator import Page as DjangoPage
+from django.core.paginator import PageNotAnInteger
+from django.core.paginator import Paginator as DjangoPaginator
+from django.db.models.query import QuerySet
+from django.http import Http404
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework_dso.embedding import ObservableIterator
+
+
+class DSOPaginator(DjangoPaginator):
+    """A paginator that supports streaming.
+
+    This paginator avoids expensive count queries.
+    So num_pages() is not supported.
+    """
+
+    def __init__(self, object_list, per_page, orphans=0, allow_empty_first_page=True):
+        super().__init__(object_list, per_page, 0, allow_empty_first_page)
```

Comment:
```suggestion
    def __init__(self, object_list, per_page, orphans=0, allow_empty_first_page=True):
        super().__init__(object_list, per_page, orphans=orphans, allow_empty_first_page=allow_empty_first_page)
```

?

Reply:
orphans is niet ondersteund. Vrij triviale feature verder. Zal ik er een check met een RuntimeWarning van maken?

Reply:
prima!